bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,184 @@
+                .unwrap()
+                .set_verify_host_callback(HostNameHandler {
+                    hostname: "localhost",
+                })
+                .unwrap(),
+        };
+
+        builder.build().unwrap()
+    }
+
+    /// Set up connections with config and custom IO
+    fn init_conn(&mut self, mode: Mode) {
+        let c_to_s_ptr = &mut self.c_to_s_buf as &mut Buffer as *mut Buffer as *mut c_void;
+        let s_to_c_ptr = &mut self.s_to_c_buf as &mut Buffer as *mut Buffer as *mut c_void;
+        let (read_ptr, write_ptr, config, conn);
+
+        match mode {
+            Mode::Client => {
+                read_ptr = s_to_c_ptr;
+                write_ptr = c_to_s_ptr;
+                config = &self.c_config;
+                conn = &mut self.c_conn;
+            }
+            Mode::Server => {
+                read_ptr = c_to_s_ptr;
+                write_ptr = s_to_c_ptr;
+                config = &self.s_config;
+                conn = &mut self.s_conn;
+            }
+        }
+
+        conn.set_blinding(Blinding::SelfService) // no blinding so benchmark time is accurate
+            .unwrap()
+            .set_config(config.clone())
+            .unwrap()
+            .set_send_callback(Some(Self::send_cb))
+            .unwrap()
+            .set_receive_callback(Some(Self::recv_cb))
+            .unwrap();
+        unsafe {
+            conn.set_send_context(write_ptr)
+                .unwrap()
+                .set_receive_context(read_ptr)
+                .unwrap();
+        }
+    }
+
+    /// Handshake step for one connection
+    fn handshake_conn(&mut self, mode: Mode) {
+        let (conn, handshaked);
+        match mode {
+            Mode::Client => {
+                info!("Client: ");
+                conn = &mut self.c_conn;
+                handshaked = &mut self.c_handshaked;
+            }
+            Mode::Server => {
+                info!("Server: ");
+                conn = &mut self.s_conn;
+                handshaked = &mut self.s_handshaked;
+            }
+        }
+        if let Ready(res) = conn.poll_negotiate() {
+            res.unwrap();
+            *handshaked = true;
+        } else {
+            *handshaked = false;
+        }
+    }
+}
+
+impl TlsImpl for S2nTls {
+    fn new() -> Self {
+        let mut new_struct = S2nTls {
+            c_to_s_buf: Box::pin(Buffer::new()),
+            s_to_c_buf: Box::pin(Buffer::new()),
+            c_config: Self::create_config(Mode::Client),
+            s_config: Self::create_config(Mode::Server),
+            c_conn: Connection::new_client(),
+            s_conn: Connection::new_server(),
+            c_handshaked: false,
+            s_handshaked: false,
+        };
+        new_struct.init_conn(Mode::Client);
+        new_struct.init_conn(Mode::Server);
+        new_struct
+    }
+
+
+
+    fn handshake(&mut self) {
+        // set limit on round trips
+        let mut iter_remaining = 10;
```

Comment:
If we know the number of roundtrips that we expect it to take, then we should just use that.